### PR TITLE
Enable nullability in BindToObject

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in BindToObject.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9554)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Binding.BindToObject.cs [hunk 1]
```diff
@@ -103,7 +104,7 @@ internal void SetBindingManagerBase(BindingManagerBase lManager)
         ///  Returns any data error info on the data source for the bound data field
         ///  in the current row
         /// </summary>
-        private string GetErrorText(object value)
+        private string GetErrorText(object? value)
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/Binding.BindToObject.cs:107]: can `text` local nullable and avoid duplicate assignment?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/Binding.BindToObject.cs:107]: Sorry, what do you mean?

File: src/System.Windows.Forms/src/System/Windows/Forms/Binding.BindToObject.cs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+            // The wait is over: DataSource is initialized.
+            _waitingOnDataSource = false;
+            _dataSourceInitialized = true;
+
+            // Rebind.
+            CheckBinding();
+        }
+
+        internal void SetBindingManagerBase(BindingManagerBase lManager)
```
Comment (halgab) [inline src/System.Windows.Forms/src/System/Windows/Forms/Binding.BindToObject.cs:80]: Are you sure `lManager` can't be null? I was under the impression the `BindingManagerBase` property whose setter calls this method can be null.

File: src/System.Windows.Forms/src/System/Windows/Forms/Binding.BindToObject.cs [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+                    }
+
+                    return type;
+                }
+
+                return FieldInfo?.PropertyType;
+            }
+        }
+
+        internal void SetValue(object value)
```
Comment (halgab) [inline src/System.Windows.Forms/src/System/Windows/Forms/Binding.BindToObject.cs:170]: Same here, I thought the argument could be null on the call site.

File: src/System.Windows.Forms/src/System/Windows/Forms/Binding.BindToObject.cs (hunk 1 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/Binding.BindToObject.cs:107]: local on line 109 can be declared as `string? text` ?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/Binding.BindToObject.cs:107]: Actually, I see that text can never be null. Maybe we should remove `?? string.Empty;` instead from line 130?

Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/Binding.BindToObject.cs:107]: final change could alter functionality. Though Interface says default is empty string, I see implementations of this interface returning null either for `errorInfo[FieldInfo.Name]` or `errorInfo.Error`.  So, i would go with declaration as `string? text ` and return `text??striong.Empty`;

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/Binding.BindToObject.cs:107]: I kept the removal of `text` and added some null checks instead.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**